### ISSUE TITLE
ES6 classes as controllers

### ISSUE THREAD

**Issue by vitaly-t** [vitaly-t/excellent#23]
It would be nice to optionally support ES6 classes as controllers, if you want to write components that require ES6 support in browsers, which becomes more of a commonplace these days.

**Minimum Signature:**

```js
class MyES6Controller extends EController {
}
```

**Full Signature:**

```js
class MyES6Controller extends EController {
    constructor(name, node) {
        super(name, node);

        // - set up public properties and methods;
        // - modify HTML, via this.node, if needed
    }

    onInit() {
    }

    onReady() {
    }

    onDestroy() {
    }
}
```

And so you can reuse the existing API:

```js
app.addController('ctrlName', MyES6Controller);
```

I have started working on branch [class] that adds such support. See also [Classes].

[class]:https://github.com/vitaly-t/excellent/tree/class
[Classes]:https://github.com/vitaly-t/excellent/wiki/Classes

**Comment by vitaly-t** [comment]
Branch [class] now has a very solid implementation of this feature, which works both on the application level and inside modules, as per [Classes].

[class]:https://github.com/vitaly-t/excellent/tree/class
[Classes]:https://github.com/vitaly-t/excellent/wiki/Classes

**Comment by vitaly-t** [comment]
In the end I decided not to focus too much on the TypeScript support, leaving the integration pain to developers. This feature is all about using native ES6 JavaScript, and that's what matters.

Released in version [1.6.0](https://github.com/vitaly-t/excellent/releases/tag/1.6.0).
